ios/Classes/VoximplantPlugin.m

```diff
@@ -100,5 +106,67 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     }
 }
 
+#pragma mark - FlutterStreamHandler
+
+- (FlutterError * _Nullable)onListenWithArguments:(id _Nullable)arguments eventSink:(nonnull FlutterEventSink)events {
+    if ([arguments isKindOfClass:[NSString class]]) {
+        NSString *type = (NSString *)arguments;
+        if ([type isEqual:@"logs"]) {
+            self.logsEventSink = events;
+        }
+    }
+    return nil;
+}
+
+- (FlutterError * _Nullable)onCancelWithArguments:(id _Nullable)arguments {
+    if ([arguments isKindOfClass:[NSString class]]) {
+        NSString *type = (NSString *)arguments;
+        if ([type isEqual:@"logs"]) {
+            self.logsEventSink = nil;
+        }
+    }
+    return nil;
+}
+
+#pragma mark - VILogDelegate
+
+- (void)didReceiveLogMessage:(NSString *)message severity:(VILogSeverity)severity {
+    if (self.logsEventSink) {
+        
+        self.logsEventSink(@{
+            @"event"               : @"onLogMessage",
+            @"level" : [self formatSeverityToString:severity],
+            @"logMessage"          : message,
+                           });
+    }
+    
+}
+
+- (NSString*)formatSeverityToString:(VILogSeverity)severity {
+    NSString *result = nil;
+    
+    switch(severity) {
+        case VILogSeverityError:
+            result = @"error";
+            break;
+        case VILogSeverityWarning:
+            result = @"warning";
+            break;
+        case VILogSeverityInfo:
+            result = @"info";
+            break;
+        case VILogSeverityDebug:
+            result = @"debug";
+            break;
+        case VILogSeverityVerbose:
+            result = @"verbose";
+            break;
+        default:
+            [NSException raise:NSGenericException format:@"Unexpected VILogSeverity."];
```

Comment:
It is better to avoid raising NSException as NSException indicates an unrecoverable issue and will crash the application. 

Since the log level mismatch is not a critical issue for the application, it's better to skip the log message in this case